charts/cryostat/tests/deployment_test.yaml

```
@@ -0,0 +1,164 @@
+suite: test deployment.yaml
+templates:
+  - deployment.yaml
+
+tests:
+  - it: should verify general deployment settings
+    asserts:
+      - equal:
+          path: kind
+          value: Deployment
+      - equal:
+          path: metadata.name
+          value: RELEASE-NAME-cryostat
+      - equal:
+          path: spec.replicas
+          value: 1
+      - equal:
+          path: spec.strategy.type
+          value: Recreate
+      - equal:
+          path: metadata.labels
+          value:
+            helm.sh/chart: cryostat-0.5.0-dev
+            app.kubernetes.io/name: cryostat
+            app.kubernetes.io/instance: RELEASE-NAME
+            app.kubernetes.io/version: "4.0.0-dev"
+            app.kubernetes.io/managed-by: Helm
+
+  - it: should validate all container settings including environment variables and security contexts
```

```suggestion
  - it: should validate all container environment variables and health probes
```

Does not seem to validate security context? There is another test case below for that.